core/docker/bin/health-check

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -euo pipefail
+
+function get_property() {
+    grep "^$1=" "$2" | cut -d'=' -f2
+}
+
+scheme=http
+port=8080
+
+config=/etc/trino/config.properties
+if [ "$(get_property 'http-server.https.enabled' "$config")" == true ]; then
+    scheme=https
+    port=$(get_property http-server.https.port "$config")
+else
+    port=$(get_property http-server.http.port "$config")
+fi
+
+endpoint="$scheme://localhost:$port/v1/info"
+
+# add --insecure to disable certificate verification in curl, in case a self-signed certificate is being used
```

Comment:
we could favor http, use it  unless `http-server.http.enabled=false`.

worth documenting the reasoning behind favoring https for the health check?

Reply:
HTTPS is not favored here, `--insecure` is always added to avoid additional condition blocks, and it'll be ignored for HTTP.

Reply:
See David's previous comment: https://github.com/trinodb/trino/pull/10413#discussion_r776240911

Reply:
HTTPS is favored in a sense that, if both endpoints are enabled, HTTPS will be used.

Reply:
Got it now, great point, thanks! I added another condition.

Reply:
I wonder should we only check if HTTP is disabled - this should imply that HTTPS must be enabled, right? Would save one call to `grep`.